src/array_api_stubs/_draft/array_object.py

```diff
@@ -239,6 +239,10 @@ def __bool__(self: array, /) -> bool:
 
         For complex floating-point operands, special cases must be handled as if the operation is implemented as the logical AND of ``bool(real(self))`` and ``bool(imag(self))``.
 
+        **Lazy implemenations**
+
+        Lazy implementations of the API standard must raise a ``ValueError`` if they are unable eagerly compute the requested value.
```

Comment:
I suggest to add the following for more context: "the Python language requires the return value to be of type `bool`, hence lazy implementations are not able to return any kind of lazy/delayed object here".